src/video.js

```diff
@@ -64,3 +66,52 @@ export const checkVideoBidSetup = hook('sync', function(bid, adUnit, videoMediaT
 
   return true;
 }, 'checkVideoBidSetup');
+
+export function registerVASTTrackers(tracker) {
+  vastTrackers.push(tracker);
+}
+
+export function insertVastTrackers(trackers, vastXml, vastImpUrl) {
+  const doc = new DOMParser().parseFromString(vastXml, 'text/xml');
+  const wrappers = doc.querySelectorAll('VAST Ad Wrapper, VAST Ad InLine');
+  try {
+    if (wrappers.length) {
+      wrappers.forEach(wrapper => {
+        trackers['impressions'].forEach(trackingUrl => {
+          const impression = doc.createElement('Impression');
+          impression.appendChild(doc.createCDATASection(trackingUrl));
+          wrapper.appendChild(impression)
+        });
+      });
+      vastXml = new XMLSerializer().serializeToString(doc);
+    }
+  } catch (error) {
+    logError('an error happened trying to insert trackers in vastXml');
+  }
+  if (vastImpUrl && vastImpUrl.length > 0) {
+    trackers['impressions'].forEach(trackingUrl => {
+      if ((trackingUrl + encodeURI(vastImpUrl)).length < 2048) {
+        vastImpUrl = trackingUrl + encodeURI(vastImpUrl);
+      }
+    });
+  };
+  return [vastXml, vastImpUrl];
+}
+
+export function getVastTrackers(bid) {
+  const videoMediaType = deepAccess(auctionManager.index.getMediaTypes(bid), 'video');
+  const context = videoMediaType && deepAccess(videoMediaType, 'context');
+  let hasTrackers = false;
+  let trackers = {'impressions': []};
+  vastTrackers.forEach(func => {
+    let tmpTrackers = func(bid);
+    for (const key in tmpTrackers) {
+      if (key in trackers && Array.isArray(tmpTrackers[key])) {
+        let replaceContext = tmpTrackers[key].map(function(url) { return url.replace('$$CONTEXT$$', context); });
```

Comment:
What's the purpose of the `$$CONTEXT$$` macro? If it's something you expect to need from your module, I think it'd be better to either retrieve it directly from your tracker function - since other trackers may need other things from the bid request.

Reply:
That's indeed something I need and I guess it might be useful for some others too. That's the reason why I put it in here instead of my tracker functions but I can remove it if you think it would better suit the way prebid works in general.

Reply:
Agree that others can find it useful - but they are likely to find many other thing useful too, so I'm worried about a growing list of macros for everything that might be needed. A better approach would be to pass the bid request to the tracker function, e.g. `registerVASTTrackers(function(bidResponse, bidRequest))`.  (the context would then be found at `bidRequest.mediaTypes.video.context`).

However that has the problem that sometimes there is no request (bidders are allowed to return "extra" bids in addition to those that were requested), so we'd need to add a third argument for the mediatypes (from `index.getMediaTypes`) and I think that can get confusing, for now I think it's better to pick what you need directly from the tracker function.

Reply:
Sure, I've updated my code accordingly